classes/SpecificPrice.php

```diff
@@ -163,7 +163,9 @@ public static function getIdsByProductId($id_product, $id_product_attribute = fa
 			FROM `'._DB_PREFIX_.'specific_price`
 			WHERE `id_product` = '.(int)$id_product.'
 			AND id_product_attribute='.(int)$id_product_attribute.'
-			AND id_cart='.(int)$id_cart);
+			AND id_cart='.(int)$id_cart.'
+                        AND id_specific_price_rule = 0'
```

Comment:
the "Catalog Price Rules" will be applied automatically when adding a new product,
it will duplicate the specific price entries from old product(including quantity specific price entry with id_specific_price_rule=0, and "Catalog Price Rules" entries with id_specific_price_rule not 0),it will also trigger the "Catalog Price Rules"  applied, so it will duplicate insert entry to specific price table,this errors happened,
so there are two way to avoid this error,
1.don't duplicate the entries for "Catalog Price Rules" from old product
2.avoid to trigger the "Catalog Price Rules" applied to the new duplicated product again

this is the first way,i am not sure it's a good solution,but i think it is a tmp way